accepted/2022/simplify-output-paths.md

```diff
@@ -0,0 +1,53 @@
+# Simplify .NET Output Paths
+
+We'd like to simplify the output paths for .NET 8.
+
+Currently, the default output path for a .NET project includes folders for the `TargetFramework` and the `RuntimeIdentifier` (if it is set).  Additionally, the publish output goes in a `publish` subfolder of the output folder.  The resulting paths are as follows:
+
+- `bin\<Configuration>\<TargetFramework>` - Build output with no `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\<RuntimeIdentifier>` - Build output with `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\publish` - Publish output with no `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\<RuntimeIdentifier>\publish` - Publish output with `RuntimeIdentifier`
+
+This makes it harder to find the output of a project, and sometimes nests one type of output in another output folder.
+
+## Proposed behavior
+
+Projects targeting .NET 8 and higher will by default not include the `TargetFramework` in the output path unless the project is multi-targeted.  Similarly, the `RuntimeIdentifier` will not by default be part of the output path.  Finally, the publish output path will be `bin\publish` by default instead of a `publish` subfolder inside of the output path.
+
+These would be the default output paths for .NET 8 projects:
+
+- `bin\<Configuration>` - Build output path for single targeted project
+- `bin\<Configuration>\<TargetFramework>` - Build output for multi-targeted project
+- `bin\publish` - Publish output for single targeted project
+- `bin\publish\<TargetFramework>` - Publish output for multi-targeted project
+
+## Implementation
+
+There are currently properties that control whether the TargetFramework and RuntimeIdentifier should be included in the output path: `AppendTargetFrameworkToOutputPath` and `AppendRuntimeIdentifierToOutputPath`.  Both of these currently default to true.  We will change the default values as follows:
+
+- `AppendTargetFrameworkToOutputPath` will default to true if the `TargetFrameworks` property is set (as multi-targeted projects still need different output paths for each TargetFramework), or the project is targeting .NET 7 or lower.  Otherwise it will default to false.
+- `AppendRuntimeIdentifierToOutputPath` will default to true only when targeting .NET 7 or lower.
+  - To be discussed: Should we also default it to true if `RuntimeIdentifiers` is set?
+
+For publish, we would change the publish output path only when targeting .NET 8 or higher and when the Configuration is set to Release (which we [plan to make the default](https://github.com/dotnet/sdk/issues/10357#issuecomment-1208564539) for .NET 8).  In that case we would set the publish output path to `$(BaseOutputPath)\publish` (by default `bin\publish`), and append the `TargetFramework` and `RuntimeIdentifier` to the path based on `AppendTargetFrameworkToOutputPath` and `AppendRuntimeIdentifierToOutputPath`.
```

Comment:
> change the publish output path only when ... Configuration is set to Release

So for Debug the same deep nesting as now would be done, thus having a inconsistent behavior in locations.
I'd publish with release always (so disable publish for Debug). At least emit a warning if published as Debug, but write to the same consistent publish-location.